Source/NSB_K.H

```diff
@@ -41,4 +41,30 @@ set_body_state_k(int i, int j, int k,
    }
 
 }
+
+AMREX_GPU_HOST_DEVICE
+inline
+void
+calc_gradp(int i, int j, int k,
+           amrex::Real                                  const space_scale,
+           amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> const dxinv,
+           amrex::Array4<const amrex::Real>             const& p,
+           amrex::Array4<amrex::Real>                   const& gp) noexcept
+{
+   gp(i,j,k,0) = space_scale * dxinv[0] * AMREX_D_TERM(  p(i+1,j  ,k  )-p(i  ,j  ,k  ),
```

Comment:
Does this part is safe in 2D? It will be looking for a k+1 value that should not exist ?

Reply:
In 2D, the `AMREX_D_TERM` expands only the first two elements that only involves k.

Reply:
AMREX_D_TERM isn't doing what's desired here, at least not for me. If I pick a point and compare the result of the D_TERM with the explicit computation, I get very different numbers.

Reply:
In both 2D and 3D ? I see a zero diff compared to the fortran in PeleLM.

Reply:
I forgot a couple of ( ) around the AMREX_D_TERM when putting thing together more cleanly, that should fix the problem.

Reply:
All looks good now.